src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs

```diff
@@ -20,48 +19,62 @@ private static Optional<object> ConvertToOptional(ConstantValue value)
             return value != null && !value.IsBad ? new Optional<object>(value.Value) : default(Optional<object>);
         }
 
-        private ImmutableArray<IOperation> ToStatements(BoundStatement statement)
+        internal ImmutableArray<BoundStatement> ToStatements(BoundStatement statement)
         {
             if (statement == null)
             {
-                return ImmutableArray<IOperation>.Empty;
+                return ImmutableArray<BoundStatement>.Empty;
             }
 
             if (statement.Kind == BoundKind.StatementList)
             {
-                return ((BoundStatementList)statement).Statements.SelectAsArray(n => Create(n));
+                return ((BoundStatementList)statement).Statements;
             }
 
-            return ImmutableArray.Create(Create(statement));
+            return ImmutableArray.Create(statement);
         }
 
         private IInstanceReferenceOperation CreateImplicitReciever(SyntaxNode syntax, ITypeSymbol type) =>
-            new InstanceReferenceExpression(InstanceReferenceKind.ImplicitReceiver, _semanticModel, syntax, type, constantValue: default, isImplicit: true);
+            new InstanceReferenceOperation(InstanceReferenceKind.ImplicitReceiver, _semanticModel, syntax, type, constantValue: default, isImplicit: true);
 
         internal IArgumentOperation CreateArgumentOperation(ArgumentKind kind, IParameterSymbol parameter, BoundExpression expression)
         {
-            var value = Create(expression);
-
             // put argument syntax to argument operation
-            var argument = value.Syntax?.Parent as ArgumentSyntax;
-
-            // if argument syntax doesn't exist, this operation is implicit
-            return new ArgumentOperation(value,
-                kind,
-                parameter,
-                s_boxedIdentityConversion,
-                s_boxedIdentityConversion,
-                semanticModel: _semanticModel,
-                syntax: argument ?? value.Syntax,
-                isImplicit: expression.WasCompilerGenerated || argument == null);
+
+            if (expression.Syntax?.Parent is ArgumentSyntax argument)
+            {
+                // if argument syntax doesn't exist, this operation is implicit
+                return new CSharpLazyArgumentOperation(this,
+                    expression,
+                    kind,
+                    s_boxedIdentityConversion,
+                    s_boxedIdentityConversion,
+                    parameter,
+                    semanticModel: _semanticModel,
+                    syntax: argument,
+                    isImplicit: expression.WasCompilerGenerated);
+            }
+            else
+            {
+                // We have to create the argument child eagerly here, as we need to use its syntax for this node, but the BoundExpression
+                // syntax may not be the correct syntax in certain scenarios (such as query clauses that need to be skipped).
```

Comment:
>syntax may not be the correct syntax in certain scenarios (such as query clauses that need to be skipped). [](start = 19, length = 106)

How did we handle this situation before?

Reply:
Previously we'd always eagerly create the child, the IArgumentOperation was not lazily instanciated at all.

---
In reply to: [237307438](https://github.com/dotnet/roslyn/pull/31010#discussion_r237307438) [](ancestors = 237307438)